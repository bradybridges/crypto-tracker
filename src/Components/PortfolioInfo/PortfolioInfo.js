import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './PortfolioInfo.scss';
import closeButton from '../../Images/close-button.svg';

class PortfolioInfo extends Component {
   renderCoinValues = () => {
     const { coinValues } = this.props;
     const elements = coinValues.map((coin) => {
       return (
         <div className="coin-value-div" key={coin.name}>
           <p>{coin.name}</p>
           <p>
            ${Number(coin.value).toFixed(2)}
           </p>
         </div>
       );
     });
     return elements;
   }

   render() {
     const { total } = this.props;
     return (
       <section id="portfolio-info-container">
         <img id="close-portfolio-btn" src={closeButton} alt="close portfolio" onClick={this.props.closePortfolio} />
         <h2 className="header">Portfolio Value</h2>
         <section>
           <h4 className="header">Total</h4>
           <p id="total-p">
            ${Number(total).toFixed(2)}
           </p>
           {this.renderCoinValues()}
         </section>
       </section>
     );
   }
}

export default PortfolioInfo;

PortfolioInfo.propTypes = {
  total: PropTypes.string.isRequired,
  coinValues: PropTypes.array.isRequired,
  closePortfolio: PropTypes.func.isRequired,
};
