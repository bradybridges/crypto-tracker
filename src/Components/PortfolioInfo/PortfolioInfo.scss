#portfolio-info-container {
  position: absolute;
  width: 90%;
  background-color: blue;
  height: 68vh;
  border-radius: 4px;
  overflow: scroll;
}

#total-p {
  margin: 0 auto;
}

.header {
  text-align: center;
  width: 100%;
}

.coin-value-div {
  display: flex;
  justify-content: center;
  align-items: center;
}