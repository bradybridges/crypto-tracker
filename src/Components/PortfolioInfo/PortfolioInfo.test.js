import React from 'react';
import PortfolioInfo from './PortfolioInfo';
import { shallow } from 'enzyme';

describe('PortfolioInfo', () => {
  let wrapper;
  const mockCoinValues = [
    {name: 'Bitcoin', value: '1321.2332'},
    {name: 'Litecoin', value: '1971.45'}
  ]
  beforeEach(() => {
    wrapper = shallow(<PortfolioInfo
      coinValues={mockCoinValues}
      total='342323.4342'
      closePortfolio={jest.fn()}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('renderCoinsValues should return an array of elements', () => {
    const result = wrapper.instance().renderCoinValues();
    const expected = [
    (
      <div>
        <p>Bitcoin</p>
        <p>$1321.23</p>
      </div>),
    (<div><p>Litecoin</p><p>$1971.45</p></div>)];
    expect(result).toEqual(expected); 
  });
});