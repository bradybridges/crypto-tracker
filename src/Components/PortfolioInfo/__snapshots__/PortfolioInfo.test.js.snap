// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioInfo should match the snapshot 1`] = `
<section>
  <img
    alt="close portfolio"
    src="close-button.svg"
  />
  <h2>
    Portfolio Value
  </h2>
  <section>
    <h4>
      Total
    </h4>
    <p>
      $
      342323.43
    </p>
    <div
      key="Bitcoin"
    >
      <p>
        Bitcoin
      </p>
      <p>
        $
        1321.23
      </p>
    </div>
    <div
      key="Litecoin"
    >
      <p>
        Litecoin
      </p>
      <p>
        $
        1971.45
      </p>
    </div>
  </section>
</section>
`;
